src/treatment.js

```diff
@@ -41,6 +41,20 @@ class Treatment extends React.Component {
   }
 }
 
+const isRender = (props, propName, componentName) => {
+  if (!(typeof props[propName] === 'function' || props.children)) {
+    return new Error(`Invalid prop \`${propName}\` or children passed to \`${componentName}\`. Expected either a render or children prop.`);
+  }
+  return null;
+};
+
+const isRenderDefaultProps = (props, propName) => {
```

Comment:
I like your idea to make sure that the value of a default prop is set based on the other prop.

Unfortunately, this doesn't work. With

```
Treatment.defaultProps = {
  render: isRenderDefaultProps,
};
```

you'll assign `render` to the function, so that the check `if (render) { ` inside treatment.js will always return true.

You could leave out default props for both `children` and `render` or use `undefined` to let anybody know that both don't have any default values.